### DISCUSSION TITLE
Chart Area changing on pan/zoom

### DISCUSSION THREAD

**Discussion by cintiapinto** [chartjs/chartjs-plugin-zoom#636] [Q&A]
Hi there!

I have a chart.js graph that takes up my entire screen with `1920x1080` resolution with a `70px` padding on the left hand side.

```    
layout: {
    padding: {
      left: 90,
    },
 },
```

When I print out the chart area when the graph is first loaded, it has the values I expect it to:

```
{
  bottom: 1080
  height: 1080
  left: 70
  right: 1920
  top: 0
  width: 1850
}
```

But when I zoom or pan in the graph using the zoom plugin, the chart area values change to:

```
{
  bottom: 1078.5
  height: 1077
  left: 70
  right: 1918.5
  top: 1.5
  width: 1848.5
}
```

This is a problem because it's causing a white border to appear around the chart, which means that the chart is then not taking up the entire screen.

I have `mode: "x"` enabled for both pan and zoom.

How can I make sure that the chart area stays set to what it's initialised to?
Is there a setting I could potentially missing or is this a known bug?

Thank you!

**Comment by kurkle** [comment]
This is probably due to scale labels changing and needing different amount of space.
I've used x-scale padding to make sure there is enough room for y-axis in all situations:

```js
options: {
  scales: {
    x: {
      afterFit(scale) {
        scale.paddingLeft = 80; // min-width for left side y-axis
      }
    }
  }
}
```
